lib/internal/modules/esm/assets/load.js

```diff
@@ -0,0 +1,241 @@
+  }
+} = globalThis;
+const {
+  codes: {
+    ERR_UNSUPPORTED_ESM_NETWORK_IMPORT_SPECIFIER,
+    ERR_UNSUPPORTED_ESM_NETWORK_BAD_RESPONSE
+  }
+} = require('internal/errors');
+const { URL } = require('internal/url');
+
+
+/**
+ * Lazily load these to avoid snapshot issues
+ */
+
+function HTTPSGet(url, opts) {
+  // eslint-disable-next-line no-func-assign
+  HTTPSGet = require('https').get;
+  return HTTPSGet(url, opts);
+}
+
+function HTTPGet(url, opts) {
+  // eslint-disable-next-line no-func-assign
+  HTTPGet = require('http').get;
+  return HTTPGet(url, opts);
+}
+
+function dnsLookup(name, opts) {
+  // eslint-disable-next-line no-func-assign
+  dnsLookup = require('dns/promises').lookup;
+  return dnsLookup(name, opts);
+}
+
+/**
+ * @param {URL} parsed
+ * @returns {Promise<CacheEntry> | CacheEntry}
+ */
+function fetchWithRedirects(parsed) {
+  const existing = getCache.get(parsed.href);
+  if (existing) {
+    return existing;
+  }
+  const handler = parsed.protocol === 'http:' ? HTTPGet : HTTPSGet;
+  const result = new Promise((fulfill, reject) => {
+    const req = handler(parsed, {
+      headers: {
+        Accept: '*/*'
+      }
+    })
+    .on('error', reject)
+    .on('response', (res) => {
+      function dispose() {
+        req.destroy();
+        res.destroy();
+      }
+      if (res.statusCode >= 300 && res.statusCode <= 303) {
+        if (res.headers.location) {
+          dispose();
+          const location = new URL(res.headers.location, parsed);
+          if (location.protocol !== 'http:' && location.protocol !== 'https:') {
+            reject(new ERR_UNSUPPORTED_ESM_NETWORK_IMPORT_SPECIFIER(
+              res.headers.location,
+              parsed.href,
+              'cannot redirect to non-network location'));
+            return;
+          }
+          return PromiseResolve(
+            fetchWithRedirects(location)
+          ).then((entry) => {
+            getCache.set(parsed.href, entry);
+            fulfill(entry);
+          });
+
+        }
+      }
+      if (res.statusCode > 303 && res.statusCode < 200) {
+        dispose();
+        reject(
+          new ERR_UNSUPPORTED_ESM_NETWORK_BAD_RESPONSE(
+            'HTTP response returned status code of ' + res.statusCode));
+        return;
+      }
+      const { headers } = res;
+      const contentType = headers['content-type'];
+      if (!contentType) {
+        dispose();
+        reject(new ERR_UNSUPPORTED_ESM_NETWORK_BAD_RESPONSE(
+          'content-type is required'));
+        return;
+      }
+      /**
+       * @type {CacheEntry}
+       */
+      const entry = {
+        resolvedHREF: parsed.href,
+        headers: {
+          'content-type': res.headers['content-type']
+        },
+        body: new Promise((f, r) => {
+          res.on('error', (e) => {
+            dispose();
+            reject(e);
+            r(e);
+          });
+          const buffers = [];
+          let size = 0;
+          res.on('data', (d) => {
```

Comment:
Given than the content is `'utf8'`, I would call `res.setEncoding('utf8')` and concatenate strings.

Reply:
I consistently have problems at scale using setEncoding, I won't change this

Reply:
It would be better to open an issue on this topic and fix those bugs instead of using a lower-performing alternative.

Reply:
this PR is not about fixing those bugs nor do i have a consistent repro. it just blips up with something around lone surrogates from time to time enough for me to not wish to use it. in the case of this usage, we always want the full body and don't have a backpressure need as well as my general assumption that if you cannot allocate 2X the memory of the source text you likely can't run the source text. if you wish to block on this, feel free but I do not want to use `.setEncoding` due to unclear problems with it in my experience. Even if it is more performant, something appears wrong with it. Working is more optimal than performance here.

Reply:
I'm not blocking on this. However a comment expressing "revisit to use `setEncoding()`" would actually be beneficial.

The latency of downloading this _could_ make an impact.

Reply:
my presumption was this was blocking since it requested changes, i misunderstood

Reply:
i also just realized with WASM this isn't always UTF8